op-conductor/conductor/flashblocks_handler.go

```diff
@@ -0,0 +1,209 @@
+		return err
+	}
+
+	return nil
+}
+
+// startRollupBoostListener initializes and starts a WebSocket client to listen for rollup boost messages
+func (oc *OpConductor) startRollupBoostListener(_ context.Context) error {
+	if oc.cfg.RollupBoostWsURL == "" {
+		oc.log.Info("rollup boost WebSocket disabled, no URL configured")
+		return nil
+	}
+
+	oc.log.Info("connecting to rollup boost WebSocket", "url", oc.cfg.RollupBoostWsURL)
+
+	// Connect to the rollup boost WebSocket
+	conn, _, err := websocket.DefaultDialer.Dial(oc.cfg.RollupBoostWsURL, nil)
+	if err != nil {
+		return fmt.Errorf("failed to connect to rollup boost WebSocket: %w", err)
+	}
+
+	oc.rollupBoostConn = conn
+	oc.log.Info("connected to rollup boost WebSocket", "url", oc.cfg.RollupBoostWsURL)
+
+	// Start a goroutine to read messages from the rollup boost WebSocket
+	go func() {
+		defer func() {
+			oc.log.Info("closing rollup boost WebSocket connection")
+			conn.Close()
+			oc.rollupBoostConn = nil
+		}()
+
+		for {
+			_, message, err := conn.ReadMessage()
+			if err != nil {
+				oc.log.Error("error reading from rollup boost WebSocket", "err", err)
+				return
+			}
+
+			oc.handleRollupBoostMessage(message)
+		}
+	}()
+
+	return nil
+}
+
+// handleRollupBoostMessage processes a message received from rollup boost
+func (oc *OpConductor) handleRollupBoostMessage(message []byte) {
+	// Only forward messages if we're the leader
+	if !oc.leader.Load() {
+		oc.log.Debug("not forwarding rollup boost message, not the leader")
+		return
+	}
+
+	// Forward the message to connected clients
+	oc.broadcastToClients(message)
+}
+
+// startWebSocketServer initializes and starts a WebSocket server
+func (oc *OpConductor) startWebSocketServer(ctx context.Context) error {
+	if oc.cfg.WebsocketServerPort <= 0 {
+		oc.log.Info("WebSocket server disabled, no port configured")
+		return nil
+	}
+
+	upgrader := websocket.Upgrader{
+		CheckOrigin: func(r *http.Request) bool {
+			return true // Allow all connections
+		},
+	}
+
+	// Create HTTP server with WebSocket endpoint
+	http.HandleFunc("/ws", func(w http.ResponseWriter, r *http.Request) {
+		conn, err := upgrader.Upgrade(w, r, nil)
+		if err != nil {
+			oc.log.Error("failed to upgrade connection", "err", err)
+			return
+		}
+
+		// Store the client connection
+		oc.wsClientMu.Lock()
+		// If we already have a connection, reject the new one
+		if oc.wsClient != nil {
+			oc.wsClientMu.Unlock()
+			oc.log.Warn("rejecting new WebSocket connection, already have one", "remote", conn.RemoteAddr())
+			conn.Close()
+			return
+		}
+		oc.wsClient = conn
+		oc.wsClientMu.Unlock()
+
+		oc.log.Info("WebSocket proxy connected", "remote", conn.RemoteAddr())
+
+		// Create a channel to detect connection errors
+		errCh := make(chan error, 1)
+
+		// Start a goroutine to read messages (just to detect disconnection)
+		go func() {
+			for {
+				_, _, err := conn.ReadMessage()
+				if err != nil {
+					errCh <- err
+					return
+				}
+			}
+		}()
```

Comment:
this function handler itself is a goroutine or connection, no need to have another go routine inside of it